aten/src/ATen/native/AutogradComposite.cpp

```diff
@@ -22,6 +22,54 @@ std::tuple<at::Tensor, at::Tensor> _unpack_dual(const at::Tensor& tensor, int64_
   return std::tuple<at::Tensor, at::Tensor>(tensor._fw_primal(level), tensor._fw_grad(level));
 }
 
+Tensor _new_zeros_with_same_feature_meta(
+    const at::Tensor& self,
+    const at::Tensor& other,
+    c10::optional<int64_t> self_num_batch_dims) {
```

Comment:
Why `self_num_batch_dims` instead of `num_feature_dims`? If we extend this to vmap(jvp later, then both self and other can be batched. In that case, in order to tell how many batch dimensions self and other have, there are two options:
1. We would need to pass in the number of batch dimensions in self and the number of batch dimensions in other
2. We would need to pass in the feature dimensions (this is the same across both tensors) and then infer the number of batch dimensions.

Reply:
Good point. Initially, it was to make it so that when we don't go through the batching rule we can just pass in zero. But I think now thats no longer relevant anyway.

Reply:
> We would need to pass in the feature dimensions (this is the same across both tensors) and then infer the number of batch dimensions.

Actually... are there any in-place operation concerns here? I'm assuming that the number of feature dimensions between the primal and the tangent must be the same, but I remember there was some discussion around weird size behavior with in-place

Reply:
Oh right, forgot about that too lol
So yeah, I guess there is not a big distinction between the two now that we will always have to pass in two values.